servers/src/test/java/tachyon/master/lineage/recompute/TestRecomputeExecutor.java

```diff
@@ -56,7 +58,8 @@ public void recomputeLauncherTest() throws Exception {
     FileSystemMaster fileSystemMaster = Mockito.mock(FileSystemMaster.class);
 
     RecomputeExecutor executor = new RecomputeExecutor(planner, fileSystemMaster);
-    executor.heartbeat();
+    // wait for the executor to finish running
+    executor.heartbeatWithFuture().get(1, TimeUnit.SECONDS);
```

Comment:
It's so complicated to use that I'd rather use it as a last resort. In this case it's much simpler to use a `Future`.